src/sidebar/components/share-annotations-panel.js

```diff
@@ -30,24 +31,24 @@ function ShareAnnotationsPanel({ analytics, toastMessenger }) {
   const store = useStoreProxy();
   const mainFrame = store.mainFrame();
   const focusedGroup = store.focusedGroup();
-
   const groupName = (focusedGroup && focusedGroup.name) || '...';
   const panelTitle = `Share Annotations in ${groupName}`;
 
-  // Generate bouncer sharing link for annotations in the current group.
+  // Not all URIs are valid to share: the document needs to be available
+  // on the web.
+  const uriShareable = mainFrame?.uri && isShareableURI(mainFrame?.uri);
+
+  // Generate bouncer sharing link.
   // This is the URI format for the web-sharing link shown in the input
-  // and is available to be copied to clipboard
-  const shareURI = ((frame, group) => {
-    return group && frame
-      ? `https://hyp.is/go?url=${encodeURIComponent(frame.uri)}&group=${
-          group.id
-        }`
-      : '';
-  })(mainFrame, focusedGroup);
+  // and is available to be copied to clipboard. We need the frame's URI
+  // (document URL) and the focused group to be available before this link
+  // can be generated.
+  const shareURI =
+    focusedGroup && mainFrame && getSharingLink(mainFrame.uri, focusedGroup.id);
 
   const copyShareLink = () => {
     try {
-      copyText(shareURI);
+      copyText(shareURI ?? '');
```

Comment:
Appeasing the TS gods

Reply:
Even though it is a bit verbose I would prefer to use a cast here (`/** @type {string} */ (shareURI)`). This avoids adding an untested path to the code and also makes it more obvious to future readers that this is purely to keep TS happy.

As an aside, this is one area where TypeScript's native syntax is much more succinct (just add an exclamation mark after the expression - `shareURI!`).